### DISCUSSION TITLE
Erase using SATA command "security-erase-enhanced"

### DISCUSSION THREAD

**Discussion by Txikilinux** [PartialVolume/shredos.x86_64#293] [Ideas]
I think it would be quicker to erase the discs using SATA command "security-erase-enhanced":

`sudo hdparm --security-erase-enhanced /dev/sda
`
I work for an NGO where we need to erase thousands of disks, I have noticed that we are limited by the SATA controller bus, which supports 400MiB/s, shared between all the disks (so, if we have only one disk, the bottleneck is the disk, but if we have ten disks, the bottleneck is the SATA controller).

Any caveats? One possible difficulty that I see is how to generate the report.

**Comment by PartialVolume** [comment]
>I think it would be quicker to erase the discs using SATA command "security-erase-enhanced":

Yes, as you say if your controller only supports 400MiB/s and that is spilt between ten discs.

Caveats:

1. The problem with only using hdparm and secure erase is like you say no automatically produced PDF certificate.
2. You could verify the disc with shredos if the secure erase results in all zeros or all ones, however some discs perform a secure erase using a pattern like 0x3F00. Currently nwipe can't perform a user specified pattern. Something that will be worth adding to nwipe's features. The downside would be that the PDF would only state the the drive was verified as zeros or ones but won't mention the method used.
3. Trust! Do you trust the drives firmware to wipe the drive without verification? See the following document that mentions how some drives don't have a good implementation of secure erase in their firmware.
[Wei SSD's and erase methods.pdf](https://github.com/user-attachments/files/17355225/Wei.SSD.s.and.erase.methods.pdf)
4. Because a whole disc erase really works that disc hard, any intermittent faults or issues will generally show up. By doing a traditional wipe you are exercising all the same data paths that a operating system would use. Why is this important? Sometimes you can get faults related to data channels in the disc electronic, by only doing a secure erase using the drives firmware you are not fully testing the drive by exercising the data channels. Whether this is important to you is purely dependant upon how sure you want to be that your discs are functioning correctly.

Most of these problems will be resolved once we integrate secure erase into nwipe so a hybrid wipe can be performed that performs a wipe using the drives firmware and then verifies the erase using a traditional block read. A prng could even be added into the method and finally generates a report that shows all the details of the wipe method and verification.
